src/mmw/js/src/draw/views.js

```diff
@@ -64,7 +65,13 @@ function validateShape(polygon) {
     var area = coreUtils.changeOfAreaUnits(turfArea(polygon), 'm<sup>2</sup>', 'km<sup>2</sup>'),
         d = new $.Deferred();
 
-    if (area > MAX_AREA) {
+    var selfIntersectingShape = turfKinks(polygon).features.length > 0;
+
+    if (selfIntersectingShape) {
+        var message = 'Unable to analyze a self-intersecting shape.'
```

Comment:
A general language comment, I think this could be improved for the user with something like:  "This watershed shape is invalid because it intersects itself.  Try drawing the shape again without crossing over its own border."